haskell/private/actions/compile.bzl

```diff
@@ -133,6 +133,11 @@ def _compilation_defaults(hs, cc, java, dep_info, srcs, import_dir_map, extra_sr
     )
 
     # Default compiler flags.
+
+    # Compilation mode.  Allow rule-supplied compiler flags to override it.
+    if hs.mode == "opt":
```

Comment:
This way `-O2` will be passed to `haddock` and `doctest`, I guess it'll be ignored and there is nothing harmful in that though?

Reply:
Good point!  I've reorganized the change so it's not passed to them; let me know what you think.